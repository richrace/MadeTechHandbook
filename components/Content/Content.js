import React from 'react';
import {View, Text} from 'react-native';

import styles from './ContentStyles/Content.style';

class Content extends React.Component {
  constructor(props) {
    super(props);
    this.state = {text: ''};
  }

  fetchData = async () => {
    const response = await fetch(
      'https://github.com/madetech/handbook/blob/master/README.md',
    );
    return await response.text();
  };

  parseLinks = data => {
    var results = [
      ...data.matchAll(new RegExp('<a href="(/madetech[^"]*?)"', 'g')),
    ].map(str => str[1]);

    return results;
  };

  async componentDidMount() {
    const rawHtml = await this.fetchData();
    const result = this.parseLinks(rawHtml);
    this.setState({text: result});
  }

  render = () => {
    return (
      <View>
        <Text style={{paddingRight: 20, paddingLeft: 20}}>
          {this.state.text}
        </Text>
      </View>
    );
  };
}

export {Content};
