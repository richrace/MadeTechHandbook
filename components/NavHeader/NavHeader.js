import React from 'react';
import { Text, Image } from 'react-native';

import styles from './NavHeaderStyles/NavHeader.style';

const madeTechLogo = require('../../images/made-tech-logo-colour-fe96d680ccf157972a92723cc5266af6.png')

class NavHeader extends React.Component {
  render = () => {
    return (
      // <Text style={styles.text}>
      //   {this.props.header}
        <Image
          style={styles.image}
          source={madeTechLogo}
        />
      // </Text>
  
    )
  }
}

export { NavHeader };