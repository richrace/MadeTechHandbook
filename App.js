/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React from 'react';

import {
  SafeAreaView,
  StatusBar,
} from 'react-native';

import {} from 'react-native/Libraries/NewAppScreen';

import {NavHeader} from './components/NavHeader/NavHeader';
import {Content} from './components/Content/Content';

const App: () => React$Node = () => {
  return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView >
        <NavHeader />
        <Content />
      </SafeAreaView>
    </>
  );
};

export default App;
