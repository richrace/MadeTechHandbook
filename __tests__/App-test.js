import 'react-native';
import React from 'react';
import App from '../App';

import {shallow} from 'enzyme';

import renderer from 'react-test-renderer';
import {Content} from '../components/Content/Content';
import {NavHeader} from '../components/NavHeader/NavHeader';

it('renders correctly', () => {
  renderer.create(<App />);
});

test('renders the Content componenet correctly', () => {
  const tree = renderer.create(<Content />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders the Content componenet correctly', () => {
  const tree = renderer.create(<NavHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Fetching the html from the madetech handbook', () => {
  it('fetches data from server when server returns a successful response', done => {
    const mockSuccessResponse = {html: 'html'};
    const mockPromise = Promise.resolve(mockSuccessResponse);
    const mockFetchPromise = Promise.resolve({
      response: () => mockPromise,
    });
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);

    const wrapper = shallow(<Content />);

    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(global.fetch).toHaveBeenCalledWith(
      'https://github.com/madetech/handbook/blob/master/README.md',
    );

    process.nextTick(() => {
      expect(wrapper.state()).toEqual({
        // need to set some state here
      });

      global.fetch.mockClear();
      done();
    });
  });
});
