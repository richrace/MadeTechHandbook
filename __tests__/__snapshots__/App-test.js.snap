// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`renders the Content componenet correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`renders the Content componenet correctly 2`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": "5%",
      "justifyContent": "center",
      "marginTop": 30,
      "opacity": 1,
      "paddingBottom": 50,
      "paddingLeft": 5,
      "paddingRight": 5,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../images/made-tech-logo-colour-fe96d680ccf157972a92723cc5266af6.png",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "height": "5%",
        "justifyContent": "center",
        "marginTop": 30,
        "paddingBottom": 50,
        "paddingLeft": 5,
        "paddingRight": 5,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`renders the Content component correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    
  </Text>
</View>
`;
